### PR TITLE
Support testing pushing c8s

### PR DESCRIPTION
**Pull request by phracek**
This pull request adds support for testing, building, and pushing varnish-container in CentOS Stream 8.
It also updates documentation.

Difference between Dockerfile.rhel8 and Dockerfile.c8s:
```bash
diff 6/Dockerfile.rhel8 6/Dockerfile.c8s
1c1
< FROM ubi8/s2i-core:1
---
> FROM quay.io/sclorg/s2i-base-c8s:c8s
20,21d19
<       com.redhat.component="varnish-6-container" \
<       com.redhat.license_terms="https://www.redhat.com/en/about/red-hat-end-user-license-agreements#rhel" \
23,24c21,22
<       usage="s2i build https://github.com/sclorg/varnish-container.git --context-dir=6/test/test-app/ rhel8/varnish-6 sample-server" \
<       name="rhel8/varnish-6" \
---
>       usage="s2i build https://github.com/sclorg/varnish-container.git --context-dir=6/test/test-app/ sclorg/varnish-6-c8s sample-server" \
>       name="sclorg/varnish-6-c8s" \
```

### CODE REVIEW

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
CentOS 7 - OpenShift 3 failed with reason, that it is not stable yet ;) https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2022-cf99083f9c

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-all]

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -38,4 +46,5 @@ jobs:
           dockerfile: ${{ matrix.dockerfile }}
           dockerfile_path: ${{ matrix.dockerfile_path }}
           tag: ${{ matrix.tag }}
+          suffix: ${{ matrix.suffix }}
```
Comment (zmiklank) [inline .github/workflows/build-and-push.yml:49]: We need to use build-and-push v2, as the suffix input is not supported in v1.

File: 6/Dockerfile.c9s [hunk 2]
```diff
@@ -17,10 +17,8 @@ LABEL summary="$SUMMARY" \
       io.k8s.display-name="Varnish 6" \
       io.openshift.expose-services="8080:http,8443:https" \
       io.openshift.tags="builder,varnish6,varnish-6" \
-      com.redhat.component="varnish-6-container" \
-      com.redhat.license_terms="https://www.redhat.com/en/about/red-hat-end-user-license-agreements#rhel" \
```
Comment (zmiklank) [inline 6/Dockerfile.c9s:21]: Could this be moved to at least a different commit, if not a different PR, stating, that the changes are in c9s, not c8s?

File: specs/multispec.yml [hunk 3]
```diff
@@ -45,7 +45,39 @@ specs:
...
+      install_pkgs: "gettext hostname nss_wrapper bind-utils varnish gcc"
+      repo_setup: yum -y module enable varnish:6  && \
+      img_name: "{{ spec.org }}/varnish-{{ spec.version }}-{{ spec.prod }}"
+      etc_path: /etc
 
   version:
     "6":
       version: "6"
+
+matrix:
```
Comment (zmiklank) [inline specs/multispec.yml:78]: excluding two distributions from the build is hidden in commit "Add c8s generation support"

Would it be possible to move it to a separate commit, and maybe explain, why we do not want to generate files for these distributions?

**Review by zmiklank — CHANGES_REQUESTED** [review]
Thank you for PR. Please see my comments.

File: 6/Dockerfile.c9s (hunk 2 above)
Comment (phracek) [reply 6/Dockerfile.c9s:21]: Everything is generated by dist-gen, I will try to reorder commit.

Comment (phracek) [reply 6/Dockerfile.c9s:21]: The general idea in the case of dist-gen is to update distgen files as one commit and add a second commit with generated files.

File: specs/multispec.yml (hunk 3 above)
Comment (phracek) [reply specs/multispec.yml]: CentOS-7 and RHEL-7 is back supported.

**Comment by phracek** [comment]
[test-all]

**Review by zmiklank — APPROVED** [review]
Thank you for the changes. LGTM.
